adagrams/game.py

```diff
@@ -1,13 +1,122 @@
+    letter_pool = get_letter_pool(letters)
+    
+    # Check each letter in the word
+    for letter in word.upper():  # Convert to uppercase
+        if letter in letter_pool and letter_pool[letter] > 0:
+            letter_pool[letter] -= 1
+        else:
+            return False
+    
+    return True
+
+# Helper function to create letter pool dictionary
+def get_letter_pool(letter_list):
+    count_dict = {}
+    for letter in letter_list:
+        if letter in count_dict:
+            count_dict[letter] += 1
+        else:
+            count_dict[letter] = 1
+    return count_dict
 
 def score_word(word):
-    pass
+    # Letter to score mapping
+    letter_scores = {
+        'A': 1, 'E': 1, 'I': 1, 'O': 1, 'U': 1, 'L': 1, 'N': 1, 'R': 1, 'S': 1, 'T': 1,
+        'D': 2, 'G': 2,
+        'B': 3, 'C': 3, 'M': 3, 'P': 3,
+        'F': 4, 'H': 4, 'V': 4, 'W': 4, 'Y': 4,
+        'K': 5,
+        'J': 8, 'X': 8,
+        'Q': 10, 'Z': 10
+    }
+    
+    total_score = 0
+    
+    # If length of the word is 7, 8, 9, or 10, then the word gets an additional 8 points
+    if len(word) >= 7:
+        total_score += 8
+        
+    for letter in word.upper():
+        if letter in letter_scores:
+            total_score += letter_scores[letter]
+        
+    return total_score
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    best_word = ""
+    highest_score = 0
+    
+    for word in word_list:
+        current_score = score_word(word)
+  
+        print(f"\nCurrent word: {word} and score: {current_score}")
+        print(f"Best word: {best_word} and highest score: {highest_score}")
+        
+        # Find best word and highest_score
+        if current_score > highest_score:
+            highest_score = current_score
+            best_word = word
+        
+        # Handle Tie breaker
+        elif current_score == highest_score:
+            current_length = len(word)
+            best_length = len(best_word)
+            
+            print("\n==== Enterting Tie Breaker ====")
+            print(f"Length of {word}: {current_length}")
+            print(f"Length of {best_word}: {best_length}")
+            
+            # Rule: one word has 10 letters
+            # If the length of the best word is not 10 but the length
+            # of the current word is 10 -> current word is the new best
+            if best_length != 10 and current_length == 10:
+                best_word = word
+            
+            # Rule: prefer the word with fewest letters
+            # If the length of the best word is not 10 AND is more than the 
+            # length of the current word -> current word is the new best
+            elif best_length != 10 and current_length < best_length:
+                best_word = word
+
+            # else best_word remains unchanged
+    
+    print(f"\nFinal best word: {best_word} and highest score: {highest_score}")
+    return (best_word, highest_score)
```

Comment:
When we return multiple values as a tuple, it is often preferred to nix the parentheses! This would give us something like:

```suggestion
    return best_word, highest_score
```